lib/extensions/ar_region.rb

```diff
@@ -176,4 +176,17 @@ def compressed_id
   def split_id
     self.class.split_id(id)
   end
+
+  def unique_within_region(field, match_case: true)
+    field_value = send(field)
+    field_matches =
+      if match_case
+        self.class.where(field => send(field))
+      else
+        self.class.where("LOWER(#{field}) = ?", field_value.downcase)
+      end
+    unless field_matches.in_region(region_id).where.not(:id => id).empty?
+      errors.add(field, "is not unique within region #{region_id}")
+    end
+  end
```

Comment:
This is simple enough, signature looks fine, and prolly apropos to include it in the extension, but did you consider adding on to the builtin validator to do this instead so you can use it with other built in validations? 

http://api.rubyonrails.org/classes/ActiveModel/Validator.html

With that, encapsulates better and combines your two validations in to one: `validates :description, :presence => true, unique_within_region => true`